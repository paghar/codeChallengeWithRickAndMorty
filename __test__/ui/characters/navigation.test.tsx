import renderer from 'react-test-renderer';
import Navigation from '../../../src/ui/characters/navigation';

describe('>>>Testing navigation', () => {
  test('+++Snapshot of navigation', () => {
    const component: any = renderer.create(
      <Navigation
        loadPage={jest.fn()}
        page={6}
        pages="12"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('+++first page no previous button', () => {
    const component: any = renderer.create(
      <Navigation
        loadPage={jest.fn()}
        page={1}
        pages="12"
      />
    );
    let tree = component.toJSON();
    expect(tree.children[0].ariaLabel).toBeUndefined();
  });
  test('+++first page no previous button', () => {
    const component: any = renderer.create(
      <Navigation
        loadPage={jest.fn()}
        page={12}
        pages="12"
      />
    );
    let tree = component.toJSON();
    expect(tree.children[6].ariaLabel).toBeUndefined();
  });
})