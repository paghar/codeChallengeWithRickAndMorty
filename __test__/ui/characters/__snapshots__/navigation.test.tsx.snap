// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Testing navigation +++Snapshot of navigation 1`] = `
<nav
  aria-label="Pagination"
  className="isolate inline-flex -space-x-px rounded-md shadow-sm"
>
  <a
    aria-label="previous"
    className="relative inline-flex items-center rounded-l-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20"
    onClick={[Function]}
  >
    <span
      className="sr-only"
    >
      Previous
    </span>
    <svg
      aria-hidden="true"
      className="h-5 w-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z"
        fillRule="evenodd"
      />
    </svg>
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    1
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    2
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    3
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    4
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    5
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    6
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    7
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    8
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    9
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    10
  </a>
  <a
    className="relative inline-flex items-center border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20 false"
    onClick={[Function]}
  >
    11
  </a>
  <a
    aria-label="next"
    className="relative inline-flex items-center rounded-r-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-gray-500 hover:bg-gray-50 focus:z-20"
    onClick={[Function]}
  >
    <span
      className="sr-only"
    >
      Next
    </span>
    <svg
      aria-hidden="true"
      className="h-5 w-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z"
        fillRule="evenodd"
      />
    </svg>
  </a>
</nav>
`;
